engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -235,6 +242,8 @@ public SlaveServer( Node slaveNode ) {
       setSslMode( true );
       this.sslConfig = new SslConfiguration( sslConfig );
     }
+
+    lock = new ReentrantReadWriteLock();
```

Comment:
As the first line of the constructor calls the zero-args constructor, you shouldn't have this line.